### ISSUE TITLE
Run selected tests => Test tree icon state: reset or remain?

### ISSUE THREAD

**Issue by rowo360** [TestCentric/testcentric-gui#1094]
**Description:**
I notice that the tree icons are handled differently in **TestCentric 2 beta** than in **TestCentric 1.6.4** when executing a set of selected tests one after the other. It's only observable when executing 'Run selected tests' not when executing 'Run all tests'.
I cannot judge exactly whether this is better or worse. But at least a conscious decision should be made, if to keep the current behavior or switch back to the old one.

TestCentric 1.6.4: All tree icons are reset to the default state before running any tests
TestCentric 2: The tree icons are not reset and therefore the icons represent an accumulate state from all previous test runs

**Reproduction:**
1. Load test project
2. Execute 'Run selected tests' for one test fixture
3. Execute 'Run selected tests' for a different test fixture
    => Observe resulting tree icons

<img width="477" alt="TreeIconState" src="https://github.com/user-attachments/assets/e1d1fddf-aba3-450e-8b0c-3ad531607258">

**Additional note:**
The Visual Studio test runner doesn't reset the icon state in this use case. Instead, it displays the icons slightly faded out in this case (see screenshot). That's also a nice solution! It might be an idea for some future improvement.

![VisualStudioTestTree](https://github.com/user-attachments/assets/732c5499-723d-4424-a8ec-8f7e5a6a6dfe)

**Environment:**
- TestCentric Version 2.0.0-beta5 and main developer branch
- OS: Windows 11
- Target Runtime for your test assembly: .NET Framework 4.8

**Comment by CharliePoole** [comment]
@rowo360 Before working on this, check the setting for "Clear Results on Reload," "Reload on Run" and "Reload on Change". Defaults may have changed...

**Comment by CharliePoole** [comment]
Regarding dimming the previously run tests, see #231. Feel free to comment there or come up with a design idea. :-)

**Comment by rowo360** [comment]
I didn't considered the 'reload' use case yet, but I expect that it becomes more complex by that.
Thanks for the hint about these settings, I checked their current values now: 
The automatic reload setting is set to 'Reload on change' and the 'Clear result on reload' is unset and disabled in the UI (with a comment about an engine reload issue). So most important the setting is not set to 'Reload on run' and overall I think that there's no reload operation involved in this issue.
And I checked that I use the identical settings using TestCentric 1.6.4 - but nontheless the behavior is different.

Finally I browsed quickly through the 1.6.4 code base - simply to see if I can find any conspicuous code lines. I spotted these lines:
![grafik](https://github.com/user-attachments/assets/95166570-0ef0-48ff-bcb2-039545715665)

**Comment by CharliePoole** [comment]
@rowo360 Technically, this isn't really blocked: we could re-enable the call to `ClearResults()` in `TestModel.cs`. But if we don't work it immediately, I think the `Blocked` label may make it easier to remember to do something about it at a later stage. What do you think?

**Comment by rowo360** [comment]
I prefer a pragmatic approach:
1. We should restore the system behavior from Version 1.6.4 . I expect that's not too much effort and it saves us from having to discuss why we have changed the behavior in the new 2.X version. After that we should close this issue. I'll try to create a PR for this issue, but nonetheless I'm fine with the 'blocked' state.

2. I think we all agree that the idea about showing different Icons for results from a previous test run than the current test run sounds nice. Therefore we should create a new issue or use an existing one (flagged as idea/feature?) to handle it. We can elaborate the idea further making notes in this issue and consider the design. Although the task sounds tempting and I'm sure to have our first successes quickly, the current tasks/issues/features should be completed first. We can't start too many tasks at the same time.

**Comment by CharliePoole** [comment]
Quick answers for now...

1. Yes, please go ahead.
2. I agree. #231 is the existing issue for this.

More... be sure to get latest and update to the latest recipe version on MyGet. Version 1.3.0 will be out in a day or two but will be the same as MyGet plus some documentation changes.
